monitoring/opentelemetry/src/test/java/io/quarkus/ts/opentelemetry/OpenTelemetryMetricsGrpcTimeoutIT.java

```diff
@@ -0,0 +1,158 @@
+package io.quarkus.ts.opentelemetry;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.http.HttpStatus;
+import org.jboss.logging.Logger;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.grpc.common.GrpcMessage;
+import io.vertx.grpc.server.GrpcServer;
+import io.vertx.junit5.VertxExtension;
+
+@Tag("https://github.com/quarkusio/quarkus/pull/47245")
+@QuarkusScenario
+@ExtendWith(VertxExtension.class)
+public class OpenTelemetryMetricsGrpcTimeoutIT {
```

Comment:
```suggestion
public class OpenTelemetryMetricsGrpcExportRecoveryIT {
```
(not testing timeouts...)